pkg/controllers/v1alpha1/datamigrate/status_handler.go

```diff
@@ -0,0 +1,181 @@
+			if err = helm.DeleteReleaseIfExists(releaseName, ctx.Namespace); err != nil {
+				m.Log.Error(err, "can't delete DataMigrate release", "namespace", ctx.Namespace, "releaseName", releaseName)
+				return err
+			}
+			return err
+		}
+		// other error
+		ctx.Log.Error(err, "can't get DataMigrate job", "namespace", ctx.Namespace, "jobName", jobName)
+		return err
+	}
+
+	if len(job.Status.Conditions) != 0 {
+		if job.Status.Conditions[0].Type == batchv1.JobFailed ||
+			job.Status.Conditions[0].Type == batchv1.JobComplete {
+			jobCondition := job.Status.Conditions[0]
+			// job either failed or complete, update DataMigrate's phase status
+			opStatus.Conditions = []datav1alpha1.Condition{
+				{
+					Type:               common.ConditionType(jobCondition.Type),
+					Status:             jobCondition.Status,
+					Reason:             jobCondition.Reason,
+					Message:            jobCondition.Message,
+					LastProbeTime:      jobCondition.LastProbeTime,
+					LastTransitionTime: jobCondition.LastTransitionTime,
+				},
+			}
+			if jobCondition.Type == batchv1.JobFailed {
+				opStatus.Phase = common.PhaseFailed
+			} else {
+				opStatus.Phase = common.PhaseComplete
+			}
+			opStatus.Duration = utils.CalculateDuration(object.GetCreationTimestamp().Time, jobCondition.LastTransitionTime.Time)
+			return nil
+		}
+	}
+
+	ctx.Log.V(1).Info("DataMigrate job still running", "namespace", ctx.Namespace, "jobName", jobName)
+	return nil
+}
+
+func (c *CronStatusHandler) UpdateStatusByHelmStatus(ctx cruntime.ReconcileRequestContext, object client.Object, opStatus *datav1alpha1.OperationStatus) error {
+	// 1. Check running status of the DataMigrate job
+	releaseName := utils.GetDataMigrateReleaseName(object.GetName())
+	cronjobName := utils.GetDataMigrateJobName(releaseName)
+	cronjob, err := utils.GetDataMigrateCronjob(c.Client, cronjobName, object.GetNamespace())
+
+	if err != nil {
+		// helm release found but cronjob missing, delete the helm release and requeue
+		if utils.IgnoreNotFound(err) == nil {
+			ctx.Log.Info("Related Cronjob missing, will delete helm chart and retry", "namespace", ctx.Namespace, "cronjobName", cronjobName)
+			if err = helm.DeleteReleaseIfExists(releaseName, ctx.Namespace); err != nil {
+				c.Log.Error(err, "can't delete DataMigrate release", "namespace", ctx.Namespace, "releaseName", releaseName)
+				return err
+			}
+			return err
+		}
+		// other error
+		ctx.Log.Error(err, "can't get DataMigrate cronjob", "namespace", ctx.Namespace, "cronjobName", cronjobName)
+		return err
+	}
+
+	jobs, err := utils.ListDataMigrateJobByCronjob(c.Client, cronjob)
+	if err != nil {
+		ctx.Log.Error(err, "can't list DataMigrate job by cronjob", "namespace", ctx.Namespace, "cronjobName", cronjobName)
+		return err
+	}
+
+	// get the newest job
+	var currentJob *batchv1.Job
+	for _, job := range jobs {
+		if job.CreationTimestamp == *cronjob.Status.LastScheduleTime {
```

Comment:
What if cronJob.Status.LastScheduleTime is different from the newest job's creation time stamp?

Reply:
You mean the newest job's creation time stamp is later than cronJob.Status.LastScheduleTime ?

Reply:
Yes. How can we make sure a job's creationTimeStamp is the same as the cronJob's lastScheduleTime? I'm not sure about the code logic in Kube-Controller-Manager, does CronJob's controller sets lastScheduleTime with the newest job's creationTimeStamp?